graphql_compiler/schema_transformation/split_query.py

```diff
@@ -0,0 +1,102 @@
+        'source_field_out_name',
+        # str, the unique out name on the @output of the the source property field in the stitch
+        'sink_field_out_name',
+        # str, the unique out name on the @output of the the sink property field in the stitch
+    )
+)
+
+
+class SubQueryNode(object):
+    def __init__(self, query_ast):
+        """Represents one piece of a larger query, targeting one schema.
+
+        Args:
+            query_ast: Document, representing one piece of a query
+        """
+        self.query_ast = query_ast
+        self.schema_id = None  # str, identifying the schema that this query targets
+        self.parent_query_connection = None
+        # SubQueryNode or None, the query that the current query depends on
+        self.child_query_connections = []
+        # List[SubQueryNode], the queries that depend on the current query
+
+
+def _get_output_directive(out_name):
+    """Return a Directive representing an @output with the input out_name."""
+    return ast_types.Directive(
+        name=ast_types.Name(value=OutputDirective.name),
+        arguments=[
+            ast_types.Argument(
+                name=ast_types.Name(value=u'out_name'),
+                value=ast_types.StringValue(value=out_name),
+            ),
+        ],
+    )
+
+
+def _get_query_document(root_vertex_field_name, root_selections):
+    """Return a Document representing a query with the specified name and selections."""
+    return ast_types.Document(
+        definitions=[
+            ast_types.OperationDefinition(
+                operation='query',
+                selection_set=ast_types.SelectionSet(
+                    selections=[
+                        ast_types.Field(
+                            name=ast_types.Name(value=root_vertex_field_name),
+                            selection_set=ast_types.SelectionSet(
+                                selections=root_selections,
+                            ),
+                            directives=[],
+                        )
+                    ]
+                )
+            )
+        ]
+    )
+
+
+def _add_query_connections(parent_query_node, child_query_node, parent_field_out_name,
+                           child_field_out_name):
+    """Modify parent and child SubQueryNodes by adding QueryConnections between them."""
+    # Create QueryConnections
+    new_query_connection_from_parent = QueryConnection(
+        sink_query_node=child_query_node,
+        source_field_out_name=parent_field_out_name,
+        sink_field_out_name=child_field_out_name,
+    )
+    new_query_connection_from_child = QueryConnection(
+        sink_query_node=parent_query_node,
+        source_field_out_name=child_field_out_name,
+        sink_field_out_name=parent_field_out_name,
+    )
+    # Add QueryConnections
+    parent_query_node.child_query_connections.append(new_query_connection_from_parent)
+    child_query_node.parent_query_connection = new_query_connection_from_child
```

Comment:
Maybe assert that the child didn't already have a parent query? Bonus points if you can figure out a convenient and efficient way to assert that the parent didn't already have this child as a child too.